test/functional/src/consoleLogger/index.js

```diff
@@ -0,0 +1,29 @@
+const createConsoleLogger = (context, logLevel) => {
+  const t = context;
+  const level = logLevel;
+  let cursor = 0;
+
+  const getNewMessages = async () => {
+    let consoleMessages = await t.getBrowserConsoleMessages();
+    consoleMessages = consoleMessages[level];
+    const newMessages = consoleMessages.slice(cursor);
+    cursor = consoleMessages.length;
+    return newMessages;
+  };
+
+  const getAllMessages = async () => {
+    let consoleMessages = await t.getBrowserConsoleMessages();
+    consoleMessages = consoleMessages[level];
+    cursor = consoleMessages.length;
```

Comment:
I'm not sure if the relationship between `getAllMessages` and `getNewMessages` is clear--specifically, that `getNewMessages` returns messages that came in since the last `getNewMessages` OR `getAllMessages` call. I think I'd have to work with it a while to know what I would expect. Anyway, I'm good with moving forward with it like it is. Thanks for doing this.